pytorch_lightning/core/hooks.py

```diff
@@ -536,9 +536,27 @@ def transfer_batch_to_device(self, batch: Any, device: Optional[torch.device] =
 
         For anything else, you need to define how the data is moved to the target device (CPU, GPU, TPU, ...).
 
+        Note:
+            This hook should only transfer the data and not modify it, nor should it move the data to
+            any other device than the one passed in as argument (unless you know what you are doing).
```

Comment:
I'm wondering, where did you find this? I recognize these docs becaue I wrote them :)

Reply:
it's down below. I just moved it up along with other notes.